### DISCUSSION TITLE
Using Avalonia in VS Code on Mac

### DISCUSSION THREAD

**Discussion by lukaszossowski** [AvaloniaUI/Avalonia#9506] [Q&A] [answered]
Hi,

I have been developing my cross platform app on Windows in VS so far but I would like to also be able to debug it and run on a Mac. 

I have basically zero experience with VS Code (and MacOS as well) so far but on Windows I just downloaded it, opened the project's folder in explorer and I could just run the app and it works. Great. I did the same on Mac but there are thousands of errors regarding "Avalonia namespace not found" etc. 

What do I need to do to run and debug it on Mac? Is there a guide to this?

**Comment by maxkatz6** [comment]
No, you don't need anything special, excluding .NET SDK installed.
If "dotnet build" works from the console, then it's likely an issue with VS Code C# plugins.

**Comment by lukaszossowski** [comment, marked as answer]
I managed to get it to work. 

dotnet restore fixed the missing avalonia namespaces but still my own projects were not detected. I had to change the urls to my project references in .csproj that were like this:

`   <ProjectReference Include="..\..\..\..\..\MyProject\MyProjectAvalonia\MyProjectAvalonia.csproj" />`

to:

`<ProjectReference Include="..\MyProjectAvalonia\MyProjectAvalonia.csproj" />`

as well as adding all project references separately (previously MyProjectAvalonia.csproj contained reference to MyProject.Backend  and referencing MyProjectAvalonia also "chained" the reference to MyProject.Backend, on Mac I had to add it separately as well to all other projects that needed it.
